src/Sylius/Component/Resource/spec/Repository/InMemoryRepositorySpec.php

```diff
@@ -0,0 +1,254 @@
+        $this->find(11)->shouldReturn(null);
+    }
+
+    function it_finds_many_objects_with_parameter(RepositableInterface $book)
+    {
+        $book->getName()->shouldBeCalled();
+
+        $this->findBy(array('name' => 'Book'))->shouldReturn(array($book));
+    }
+
+    function it_finds_all_objects_when_parameter_is_not_set(RepositableInterface $book, RepositableInterface $shirt)
+    {
+        $this->findBy()->shouldReturn(array($book, $shirt));
+    }
+
+    function it_finds_many_objects_with_parameter_order_limit_offset(
+        RepositableInterface $secondBook,
+        RepositableInterface $thirdBook,
+        RepositableInterface $fourthBook
+    ) {
+        $secondBook->getId()->willReturn(80);
+        $thirdBook->getId()->willReturn(81);
+        $fourthBook->getId()->willReturn(82);
+
+        $secondBook->getName()->willReturn('Book');
+        $thirdBook->getName()->willReturn('Book');
+        $fourthBook->getName()->willReturn('Book');
+
+        $secondBook->getRating()->willReturn(3);
+        $thirdBook->getRating()->willReturn(2);
+        $fourthBook->getRating()->willReturn(1);
+
+        $this->add($secondBook);
+        $this->add($thirdBook);
+        $this->add($fourthBook);
+
+        $this->findBy(
+            array('name' => 'Book'),
+            array('rating' => 'ASC'),
+            $limit = 2,
+            $offset = 1
+        )->shouldReturn(array($thirdBook, $secondBook));
+    }
+
+    function it_throws_unexpected_value_exception_on_multiple_results_while_finding_one_object_by_parameter(RepositableInterface $secondBook)
+    {
+        $secondBook->getId()->willReturn(81);
+        $secondBook->getName()->willReturn('Book');
+
+        $this->add($secondBook);
+
+        $this->shouldThrow(\UnexpectedValueException::class)->during('findOneBy', array(array('name' => 'Book')));
+    }
+
+    function it_finds_one_object_by_parameter(RepositableInterface $book)
+    {
+        $this->findOneBy(array('name' => 'Book'))->shouldReturn($book);
+    }
+
+    function it_finds_all_objects_in_memory(RepositableInterface $book, RepositableInterface $shirt)
+    {
+        $this->findAll()->shouldReturn(array($book, $shirt));
+    }
+
+    function it_return_empty_array_when_memory_is_empty(RepositableInterface $book, RepositableInterface $shirt)
+    {
+        $this->remove($book);
+        $this->remove($shirt);
+
+        $this->findAll()->shouldReturn(array());
+    }
+
+    function it_creates_paginator()
+    {
+        $this->createPaginator()->shouldHaveType(Pagerfanta::class);
+    }
+
+    function it_returns_stated_class_name()
+    {
+        $this->getClassName()->shouldReturn(RepositableInterface::class);
+    }
+}
+
+class Void
+{
+}
+
+interface RepositableInterface extends ResourceInterface
+{
+    /**
+     * @return string
+     */
+    public function getName();
+
+    /**
+     * @return int
+     */
+    public function getRating();
+}
+
+class Repositable implements RepositableInterface
```

Comment:
Classes that are uses for spec should be created in `.../spec/Fixtures/` (see https://github.com/pjedrzejewski/Sylius/blob/introduce-resource-factories/src/Sylius/Component/Inventory/spec/Fixtures/SampleStockable.php).